### PR TITLE
Optimize compare to `HexConverter.FromChar` result

### PR DESCRIPTION
**Pull request by xtqqczze**
`ldc.i4.s 127` is 2 bytes of IL vs 5 bytes for `ldc.i4 255`

`cmp ecx, 0x7f` is 3 bytes while `cmp ecx, 0xff` is 6 bytes

[sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs4fsNKTuAQIuOgFlSACmwA7DHWwBKNhy6rCAdnl0APAF46ABgTqAYgG4VnAL6XuDPgKEjRBqbPlL2lVWs3Y6AIT6RibmtjaUVkA===)

### CODE REVIEW

**Review by dakersnar — APPROVED** [review]
This looks good to me, although it will need a second review. As a side note, I'm curious if Roslyn has an "instruction selection" step that could handle this optimization automatically, although I could see this being too specific to be worth it.

**Comment by tannergooding** [comment]
Roslyn is not an "optimizing compiler" and doesn't really do any in depth transformations.

These type of "saving 1-byte of IL" here or there are also typically not impactful. There are some specific edge cases where "less IL" means an inline that otherwise would've failed will now pass, but it's not in general an issue.

**Comment by tannergooding** [comment]
Also notably, Roslyn could not do these transformations implicitly. `(x | y) == 0xFF` being transformed into `(x | y) > 0x7F` is not equivalent.

It works here, in these scenarios, given the domain constraint that `FromChar` will return a valid hex character (`0-9`, `A-F`, or `a-f`) or `0xFF`.

**Comment by stephentoub** [comment]
I agree with @tannergooding.

**Comment by tannergooding** [comment]
Given that, this isn't one we'll be taking.

We'd like to thank you for the contribution @xtqqczze and would welcome other changes/contributions in the future, particularly if you can provide numbers showing measurable perf improvements

**Pull request closed without merging**
